lib/entities/HardQuoteRequest.ts

```diff
@@ -122,12 +121,12 @@ export class HardQuoteRequest {
         ? TradeType.EXACT_INPUT
         : TradeType.EXACT_OUTPUT
     } 
-    // else if (this.order instanceof UnsignedV3DutchOrder) {
-    //   const startAmount = this.order.info.input.startAmount;
-    //   return startAmount.eq(V3DutchOrderBuilder.getMinAmountOut(startAmount, this.order.info.input.curve.relativeAmounts))
-    //     ? TradeType.EXACT_INPUT
-    //     : TradeType.EXACT_OUTPUT
-    // } 
+    else if (this.order instanceof UnsignedV3DutchOrder) {
+      const startAmount = this.order.info.input.startAmount;
+      return startAmount.eq(V3DutchOrderBuilder.getMinAmountOut(startAmount, this.order.info.input.curve.relativeAmounts))
```

Comment:
`this.order.info.input.curve` could be undefined so we can just add that in the check
`!this.order.info.input.curve || startAmount.eq(V3DutchOrderBuilder.getMinAmountOut(startAmount, this.order.info.input.curve.relativeAmounts))`